index.bs

```diff
@@ -568,13 +578,23 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered sets=].
 
-1. For each |authenticator| currently available on this platform, if
-    <code>|options|.{{MakeCredentialOptions/attachment}}</code> is [=present|not present=] or its value matches
-    |authenticator|'s [=attachment modality=], [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|. 
-
 1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is 
     "{{NotFoundError}}", and terminate this algorithm.
 
+1. For each [=authenticator=] currently available on this platform, process each option in 
+    |options|.{{MakeCredentialOptions/authenticatorSelection}} by checking if the [=authenticator=] satisfies the 
+    requirement of the option. 
+        1. Check if |attachment| is [=present|not present=] or its value matches
+            |authenticator|'s attachment modality, [=continue=].
+        1. If |requireResidentKey| is set to |true|, check if the |authenticator| is capable of storing 
```

Comment:
> Also, I wonder if this filter should be happening on the CTAP side.

yes, something to wonder about and is it not a similar question to this one (and the one following it):  https://github.com/w3c/webauthn/pull/427#discussion_r114213505  ?

WRT CTAP, at this point in time, Section 4 "authenticator API" of the published CTAP spec is stable (tho there may be outstanding proposals to alter it): https://fidoalliance.org/specs/fido-v2.0-rd-20161004/fido-client-to-authenticator-protocol-v2.0-rd-20161004.html#authenticator-api